tools/digester/main.go

```diff
@@ -274,13 +276,29 @@ func readOneDigest(ctx context.Context, image *Image, index int, wg *sync.WaitGr
 	imgRef, err := docker.ParseReference(fullName)
 	exitOnError(err, "failed to parse container reference")
 
-	digest, err := docker.GetDigest(ctx, nil, imgRef)
+	digest, err := retryGetDigest(5, 1, ctx, nil, imgRef)
+
 	exitOnError(err, "failed to get digest from image")
 
 	ch <- message{index: index, digest: digest.Hex(), fullName: fullName}
 	wg.Done()
 }
 
+func retryGetDigest(attempts int, sleep time.Duration, ctx context.Context, sys *types.SystemContext, imgRef types.ImageReference) (digest digest.Digest, err error) {
```

Comment:
please move ctx to be the first parameter
```suggestion
func retryGetDigest(ctx context.Context, attempts int, sleep time.Duration, sys *types.SystemContext, imgRef types.ImageReference) (digest digest.Digest, err error) {
```

Reply:
done, thanks